code/src/compononets/TaskForm.js

```diff
@@ -0,0 +1,40 @@
+import React, { useState } from "react";
+import { useDispatch } from "react-redux";
+import uniqid from "uniqid";
+
+import tasks from "../reducers/tasks";
+
+const TaskForm = () => {
+  const [value, setValue] = useState("");
+  const dispatch = useDispatch();
+
+  const onFormSubmit = (e) => {
+    e.preventDefault();
+
+    const newTask = {
+      id: uniqid(),
+      text: value,
+      complete: false,
+    };
+
+    dispatch(tasks.actions.addTask(newTask));
+    setValue("");
+  };
+
+  return (
+    <form onSubmit={onFormSubmit} className="form">
+      <input
+        className="input"
+        type="text"
+        value={value}
+        onChange={(e) => setValue(e.target.value)}
+        placeholder=" +  Add task"
+      />
+      <button type="submit" className="submit-button">
+        <i className="fas fa-plus"></i>
+      </button>
```

Comment:
👍

Reply:
In this picture, you can see that you need to prevent content from your input go under your add button and make it impossible to send an empty string to your reducer. 
![image](https://user-images.githubusercontent.com/50926159/116784720-13dd0400-aa96-11eb-8e1b-896053e514f1.png)

Reply:
Done